uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/absence-tab/components/absence-tab-entitlements.component.js

```diff
@@ -11,16 +12,53 @@ define([
       return settings.pathTpl + 'components/absence-tab-entitlements.html';
     }],
     controllerAs: 'entitlements',
-    controller: ['$log', controller]
+    controller: AbsenceTabEntitlementsController
   });
 
-  function controller ($log) {
+  AbsenceTabEntitlementsController.$inject = ['$q', '$log', 'AbsenceType',
+    'OptionGroup'];
+
+  function AbsenceTabEntitlementsController ($q, $log, AbsenceType, OptionGroup) {
     $log.debug('Component: absence-tab-entitlements');
 
-    var vm = {};
+    var calculationUnits = [];
+    var vm = this;
+
+    vm.absenceTypes = [];
+    vm.loading = { component: true };
+
+    (function init () {
+      $q.all([
+        loadAbsenceTypes(),
+        loadCalculationUnits()
```

Comment:
AbsenceType.loadCalculationUnits() (a method of the Absence Type model) automatically loads and populate calculation units. Is there any reason why it was not used like intended?

```js
AbsenceType.all()
.then(AbsenceType.loadCalculationUnits) // it will accept returned absence types, load units and map it
.then(function (absenceTypes) { .... });
```